core/src/env2/srml/impls.rs

```diff
@@ -349,13 +349,13 @@ where
         ext::deposit_event(topics, data);
     }
 
-    fn invoke_runtime<O, V>(buffer: &mut O, call_data: &V)
+    fn invoke_runtime<O, V>(buffer: &mut O, call_data: V)
```

Comment:
Since we are just encoding the `call_data` parameter which takes a `&self` we do not need to take over ownership of anything. So please change it back to just `&<Self as EnvTypes>::Call`. So we can remove the `V` generic parameter as far as I can see.
Besides this looks good.

Reply:
Sure thing. I just followed the old convention - figured having it implement Into might fit a general case I hadn't considered.